lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -69,6 +85,54 @@ export default function AssignmentActivity() {
           {assignment.data && title}
         </h2>
       </div>
+      <DashboardActivityFilters
+        selectedCourseIds={courseId ? [courseId] : []}
```

Comment:
When the assignment is loading, we set the selected course IDs to an empty set, but that's not quite accurate. In that state we know there is a selected course ID, but we don't know what it is. We don't want to load the list of assignments in all courses, and then once we have the course ID, load the filtered list of assignments.

The dumbest solution here would be to avoid rendering the `DashboardActivityFilters` component unless we have a course ID. This will lead to some flicker, so this could be improved if this component was able to communicate down to `DashboardActivityFilters` that it is still loading the current selection.